src/test/java/persistence/sql/dml/builder/DeleteQueryBuilderTest.java

```diff
@@ -0,0 +1,90 @@
+package persistence.sql.dml.builder;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import database.DatabaseServer;
+import database.H2;
+import java.lang.reflect.Field;
+import java.sql.SQLException;
+import java.util.List;
+import jdbc.JdbcTemplate;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import persistence.dialect.h2.H2Dialect;
+import persistence.meta.MetaData;
+import persistence.sql.ddl.builder.CreateQueryBuilder;
+import persistence.sql.fixture.PersonFixtureStep3;
+import persistence.sql.fixture.PersonInstances;
+
+@DisplayName("4. 요구사항 DELETE 구현하기")
+public class DeleteQueryBuilderTest {
+
+  private static DatabaseServer server;
+  private static Class<PersonFixtureStep3> person;
+  private static JdbcTemplate jdbcTemplate;
+
+  @BeforeAll
+  static void setup() throws SQLException {
+    person = PersonFixtureStep3.class;
+    server = new H2();
+    server.start();
+
+    MetaData meta = MetaData.of(person, new H2Dialect());
+    CreateQueryBuilder createQueryBuilder = new CreateQueryBuilder();
+    String query = createQueryBuilder.createCreateQuery(meta);
+
+    jdbcTemplate = new JdbcTemplate(server.getConnection());
+    jdbcTemplate.execute(query);
+
+    InsertQueryBuilder<PersonFixtureStep3> insertQueryBuilder = new InsertQueryBuilder();
+    String queryFirst = insertQueryBuilder.createInsertQuery(PersonInstances.첫번째사람);
+    String querySecond = insertQueryBuilder.createInsertQuery(PersonInstances.두번째사람);
+    jdbcTemplate.execute(queryFirst);
+    jdbcTemplate.execute(querySecond);
+  }
+
+  @AfterAll
+  static void teardown() {
+    server.stop();
+  }
+
+  @Test
+  @DisplayName("Delete SQL 구문을 생성합니다.")
+  public void deleteDMLfromEntity() throws NoSuchFieldException, IllegalAccessException {
+    DeleteQueryBuilder<PersonFixtureStep3> deleteQueryBuilder = new DeleteQueryBuilder<>();
+    Field field = PersonInstances.첫번째사람.getClass().getDeclaredField("id");
+    field.setAccessible(true);
+    field.set(PersonInstances.첫번째사람, 1L);
+
+    String query = deleteQueryBuilder.createDeleteQuery(PersonInstances.첫번째사람, field);
+
+    assertThat(query).isEqualTo("DELETE FROM USERS WHERE id = 1;");
+  }
```

Comment:
```suggestion
  public void deleteDMLfromEntity() throws NoSuchFieldException, IllegalAccessException {
    DeleteQueryBuilder deleteQueryBuilder = new DeleteQueryBuilder<>();

    String actual = deleteQueryBuilder.createDeleteQuery("USERS", "id", 1L);

    assertThat(actual).isEqualTo("DELETE FROM USERS WHERE id = 1;");
  }
```
```java
public class DeleteQueryBuilder {
  private static final String DELETE_SQL_QUERY = "DELETE FROM %s WHERE %s = %s;";

  public String createDeleteQuery(String targetTableName, String targetName, Object targetValue) {
    return String.format(DELETE_SQL_QUERY, targetTableName, targetName, targetValue);
  }
}
```

TDD 방법론으로 아주 작은 단계부터 쌓아보면 아주 단순한 구조가 도출 될거라고 생각되어집니다.

테스트 하고 싶은 본질이 무엇인지!
만들고자 하는 객체에게 어떠한 메시지를 전달하고 싶은지!
아주 간단한 구조부터 고민을 시작해보면 어떨까요!?

추가되는 요구사항에 대해서 로직을 붙이면서 복잡해진다면!
그때 `객체`,`추상화`를 고민해보아도 좋을것 같아요! 😄

Reply:
![스크린샷 2023-10-28 오후 10 21 42](https://github.com/next-step/jpa-query-builder/assets/12059592/9086a7b3-8702-4ec0-a34d-e2bdd310ff4f)

Reply:
이 부분에서 외부 구현체에서 queryBuilder를 사용하게 될때 정확히 어떤 컨텍스트를 가지고 사용할 지 몰라서

구현이 조금 추상화 됐던걸로 보입니다.
`아주 작은 단계부터 쌓아보면 아주 단순한 구조가 도출 될거라고 생각되어집니다.`
해당 말씀을 조금 새겨담아서 조금 더 단순한 구조를 도출할 수 있게 만들어 보겠습니다!

Reply:
EntityManager를 진행하면서 어떤 말씀인지 이해했습니다.

`빌더가 SQL 쿼리만 구현할 수 있게끔 객체를 받지말아라` 로 이해가 됐습니다!